python/core-sdk/alation_ai_agent_sdk/tools.py

```diff
@@ -565,6 +570,203 @@ def run(
+        """
+        Filter each field to include only the 8 required properties.
+
+        Args:
+            raw_fields: Full field objects from the API
+
+        Returns:
+            List of field objects with only the 8 selected properties
+        """
+        filtered_fields = []
+
+        # The 8 properties we want to keep
+        selected_properties = [
+            'id',
+            'name_singular',
+            'field_type',
+            'allowed_otypes',
+            'options',
+            'tooltip_text',
+            'allow_multiple',
+            'name_plural'
+        ]
+
+        for field in raw_fields:
+            filtered_field = {}
+            for prop in selected_properties:
+                filtered_field[prop] = field.get(prop)
+            filtered_fields.append(filtered_field)
+
+        return filtered_fields
+
+
+class GetDataDictionaryInstructionsTool:
+    """
+    Generates comprehensive instructions for creating Alation Data Dictionary CSV files.
+
+    This tool provides LLMs with complete formatting rules, validation schemas, and examples
+    for transforming object metadata into properly formatted data dictionary CSVs.
+    """
+
+    def __init__(self, api: AlationAPI):
+        self.api = api
+        self.name = self._get_name()
+        self.description = self._get_description()
+
+    @staticmethod
+    def _get_name() -> str:
+        return "get_data_dictionary_instructions"
+
+    @staticmethod
+    def _get_description() -> str:
+        return """
+        Generates comprehensive instructions for creating Alation Data Dictionary CSV files.
+
+        Automatically fetches current custom field definitions and provides:
+        - Complete CSV format specifications with required headers
+        - Custom field formatting rules and validation schemas
+        - Object hierarchy grouping requirements
+        - Field-specific validation rules and examples
+        - Ready-to-use transformation instructions for LLMs
+
+        WHEN TO USE:
+        - Before generating data dictionary CSV files for bulk metadata upload
+        - To understand proper formatting for different object types and custom fields
+        - When transforming catalog objects and metadata into upload-ready format
+
+        WORKFLOW:
+        1. Call this tool to get comprehensive formatting instructions
+        2. Use the instructions to transform your object data into properly formatted CSV
+        3. Upload the CSV file to Alation using the Data Dictionary interface
+
+        OBJECT HIERARCHY REQUIREMENTS:
+        - RDBMS objects (data, schema, table, attribute) must be in ONE CSV file together
+        - BI objects (bi_server, bi_folder, bi_datasource, bi_datasource_column, bi_report, bi_report_column) need separate CSV
+        - Documentation objects (glossary_v3, glossary_term) need separate CSV
+        - Title field is NOT supported for BI objects (read-only from source system)
```

Comment:
This feels more like a part of the instruction than the description used for tool discovery. Any reason why you added this?